api/pre_process_data.py

```diff
@@ -73,6 +77,9 @@ def __preprocess_data(
         data[indicator1] = pd.to_numeric(data[indicator1], errors="coerce")
         data[indicator2] = pd.to_numeric(data[indicator2], errors="coerce")
 
+        # Filter data by years starting from self.from_year
```

Comment:
Always filter as soon as you can. So after reformatting the years (L72-L73).

Reply:
Also, kinda related, try to figure out how to do the country filtering as soon as possible (within this function).